# Exam 1 Answers Miguel Rey Romero

# Problem 2

In [1]:
ls -lh

total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  6 09:52  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  6 09:52  DISCLAIMER
-rw-rw-r-- 1 osboxes osboxes    0 Sep  6 11:07  Exam_1_Answer
-rw-rw-r-- 1 osboxes osboxes 1.4K Sep  6 11:20  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  6 09:52 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  6 09:52 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  6 09:52  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  6 09:52  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  6 09:52  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  6 09:52  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  6 09:52  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  6 09:52  README.md


### ls print the content of the directory.
### The permissions of that file are: 
#### "rw" (read and write) to users.
#### "rw" (read and write) to groups.
#### "r" (read) to anyone.

# Problem 3

In [5]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


### The command head prints first 10 lines, the argument -1 prints only the first line.

# Problem 4

In [6]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


### The command wc (word count) allows for different counts, the argument -l count the lines. The Locus_Germplasm_Phenotype_20130122.txt have 7216 lines.

# Problem 5

In [1]:
tail +2 Locus_Germplasm_Phenotype_20130122.txt > Data_Only.csv

In [2]:
wc -l Data_Only.csv

7215 Data_Only.csv


### As we have removed the header, there must be one less line. As there are 7215, it is shown that we have removed a line.
### To check that the deleted line is the header, i use the command cat Data_Only.csv, i dont show the putput because its too large.

# Problem 6

In [ ]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt

### The command grep print the lines that includes the word root, the argument -i make insensitive to cases, and the argument -a assumes that the source of the file is going to be Binary, because i dont know the original source of this file. I dont show the output because its too large.

# Problem 7

In [21]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt | grp -E -oa 'AT[1-5]G[0-9]{5}' > Root-associated-Loci.txt

### To search the phenotype including the word root i use the same command used in the previous exercise, then i use a pipe with other grep to print the AGI codes, this grep command is used with the argument -o (print the matches not the lines) and with the regular expresion 'AT[1-5]g[0-9]{5}', [1-5] because the number of chromosomes are between 1 and 5, and [0-9]{5} because after the g there are 5 numbers between 0 and 9. Then the strout is redirected to a file called Root-associated-Loci.txt

# Problem 8

In [29]:
grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt | grep -E -oa '[0-9]+$' > Root-associated-Publications.txt

### The first grep is the same of the previous exercises, the second grep is similar to the previous exercise but change the regular expresion, in this case contains numbers between 0 and 9, i use the symbol + because the lenght of the PID is uknown, the "$" is used because the ID is at the end of the line.

# Problem 9

In [53]:
 grep -ia "root" Locus_Germplasm_Phenotype_20130122.txt |grep -E -o '[1-5]'| sort | uniq -c

   3373 1
   1978 2
   1746 3
   1242 4
   1621 5


### As the genes of interest are those associated with root, i use the grep command for select this genes, to avoid the repeated chromosomes i use the command uniq, i use this command with the argument -c to print the numbers of genes are in each chromosome. The command sort its necesary to use the command uniq.

This work has been done with Sara Temiño Macaya because we are a couple and we live together. We have also delivered it late because we were travelling, and there has been a cut in the A3 road which has delayed us a little more. We are sorry for the inconvenience, thank you for your attention.